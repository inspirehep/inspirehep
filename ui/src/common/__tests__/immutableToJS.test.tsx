import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import {
  convertAllImmutablePropsToJS,
  convertSomeImmutablePropsToJS,
} from '../immutableToJS';

function MutableDummy(props) {
  return <span {...props} />;
}

describe('immutableToJS', () => {
  describe('convertAllImmutablePropsToJS', () => {
    it('converts all immutable props to built in js', () => {
      const ImmutableDummy = convertAllImmutablePropsToJS(MutableDummy);
      const immutableProp = fromJS({
        list: [{ foo: 'bar1' }, { foo: 'bar2' }],
      });
      const primitiveProp = 'string';
      const mutableProp = {
        array: [{ foo: 'bar' }],
      };
      const wrapper = shallow(
        <ImmutableDummy
          immutableProp={immutableProp}
          mutableProp={mutableProp}
          primitiveProp={primitiveProp}
        />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('convertSomeImmutablePropsToJS', () => {
    it('converts some immutable props to built in js', () => {
      const ImmutableDummy = convertSomeImmutablePropsToJS(MutableDummy, [
        'immutableProp1',
      ]);

      const immutableProp1 = fromJS({
        list: [{ foo: 'bar1' }, { foo: 'bar2' }],
      });
      const immutableProp2 = fromJS({
        foo: 'bar',
      });
      const primitiveProp = 'string';
      const mutableProp = {
        array: [{ foo: 'bar' }],
      };
      const wrapper = shallow(
        <ImmutableDummy
          immutableProp1={immutableProp1}
          immutableProp2={immutableProp2}
          mutableProp={mutableProp}
          primitiveProp={primitiveProp}
        />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
