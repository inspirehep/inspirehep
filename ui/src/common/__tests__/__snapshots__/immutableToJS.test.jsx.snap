// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`immutableToJS convertAllImmutablePropsToJS converts all immutable props to built in js 1`] = `
<MutableDummy
  immutableProp={
    Object {
      "list": Array [
        Object {
          "foo": "bar1",
        },
        Object {
          "foo": "bar2",
        },
      ],
    }
  }
  mutableProp={
    Object {
      "array": Array [
        Object {
          "foo": "bar",
        },
      ],
    }
  }
  primitiveProp="string"
/>
`;

exports[`immutableToJS convertSomeImmutablePropsToJS converts some immutable props to built in js 1`] = `
<MutableDummy
  immutableProp1={
    Object {
      "list": Array [
        Object {
          "foo": "bar1",
        },
        Object {
          "foo": "bar2",
        },
      ],
    }
  }
  immutableProp2={
    Immutable.Map {
      "foo": "bar",
    }
  }
  mutableProp={
    Object {
      "array": Array [
        Object {
          "foo": "bar",
        },
      ],
    }
  }
  primitiveProp="string"
/>
`;
